### DISCUSSION TITLE
How to trigger an action on error when using handleSubmit ?

### DISCUSSION THREAD

**Discussion by MatthieuBrouillard** [react-hook-form/react-hook-form#9165] [Q&A] [answered]
Hi,

We would like to trigger some special actions when when try to submit a form and that form has an error. For example, displaying an error message in a notification or something like that. The issue we are facing is that since the `handleSubmit` function does not call its callback function when there are errors in the `formState`, we can't add any error managing logic in that callback function.

How could we build it in a way that we can manage the error state ourselves instead of blocking it by default in `handleSubmit` ?

Some code example :

```
const Form = () => {
  const {
    reset: resetForm,
    handleSubmit,
    formState: { isDirty: isFormDirty, errors },
  } = useForm()

  const handleUpdate = (formValues) => {
    if (!isEmpty(errors) {
      // We are not able to reach this since this is blocked at the handleSubmit level
      showErrorNotification();
    } else {
      sendDataToServer(formValues)
    }
    });
  };

  return (
   <form onSubmit={handleSubmit(handleUpdate)}>
     {formFields}
   </form>
  )
}
```

Thanks for your help, feel free to tell me if you need anything other information!

**Comment by leapful** [comment, marked as answer]
@MatthieuBrouillard 

You can use "onError" callback of "handleSubmit".

For more detail, please refer to this topic: https://github.com/react-hook-form/react-hook-form/discussions/9091
